store/src/java/com/zimbra/cs/service/formatter/ArchiveFormatter.java

```diff
@@ -957,6 +958,8 @@ public void saveCallback(UserServletContext context, String contentType, Folder
                         }
                         try {
                             id = new ItemData(readArchiveEntry(ais, aie));
+                        } catch (EOFException e) {
+                            throw ServiceException.FAILURE("Error reading file", e);
```

Comment:
Is there any particular reason to catch not the IOException but only the EOFException?  The "readArchiveEntry" may throw some other subclass of IOException which would also cause some endless loop.

Reply:
Done